### PR TITLE
[google_sign_in] Handle new style URLs in GoogleUserCircleAvatar

### PR DESCRIPTION
**Pull request by ditman**
## Description

Change the way sizing/cropping information is added to the Identity photoUrl used to render the widget, depending on the URL format.

## Related Issues

* https://github.com/flutter/flutter/issues/44024

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/google_sign_in/lib/src/fife.dart [hunk 1]
```diff
@@ -0,0 +1,51 @@
+/// A regular expression that matches against the "size directive" path
```
Comment (tvolkert) [inline packages/google_sign_in/lib/src/fife.dart]: This file needs a copyright header

File: packages/google_sign_in/lib/src/fife.dart [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+/// image, and "`c`" indicates we want the image cropped.
+final RegExp sizeDirective = RegExp(r'^s[0-9]{1,5}(-c)?$');
+
+/// Adds [size] directive to [photoUrl].
+///
+/// Depending on the format of photoUrl, this might:
+///  * Insert information as the last path segment (old style URLs), before
+///    the image filename. The format is described in [sizeDirective].
+///  * Insert information at the end of the URL, after an = sign. The format
+///    is described within this method.
```
Comment (tvolkert) [inline packages/google_sign_in/lib/src/fife.dart]: Can you describe the format in the docs?

File: packages/google_sign_in/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,7 @@
+## 4.0.12
+
+* Fix GoogleUserCircleAvatar to handle new style profile image URLs.
```
Comment (collinjackson) [inline packages/google_sign_in/CHANGELOG.md]:
```suggestion
* Fix `GoogleUserCircleAvatar` to handle new style profile image URLs.
```

**Comment by ditman** [comment]
Thanks for taking a look! I'll rebase and incorporate all the changes suggested above!

File: packages/google_sign_in/lib/src/fife.dart (hunk 1 above)
Comment (ditman) [reply packages/google_sign_in/lib/src/fife.dart]: Yep, it does. Added!

File: packages/google_sign_in/CHANGELOG.md (hunk 3 above)
Comment (ditman) [reply packages/google_sign_in/CHANGELOG.md]: Yep, done!

**Comment by ditman** [comment]
Rebased with master, and addressed PR comments. Waiting for automated checks to pass before landing this.

File: packages/google_sign_in/test/fife.dart [hunk 4]
```diff
@@ -0,0 +1,66 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
```
Comment (ditman) [inline packages/google_sign_in/test/fife.dart]: This file needs to be called `fife_test.dart` for tests to run automatically.

**Comment by ditman** [comment]
(Force-push to trigger re-analysis)

**Pull request merged**